Source/SplineFollow/Private/Components/SFSplineComponent.cpp

```diff
@@ -59,6 +88,41 @@ void USFSplineComponent::UpdateSpline()
     }
 }
 
+void USFSplineComponent::Serialize( FArchive & archive )
+{
+    if ( archive.IsSaving() )
+    {
+        for ( const auto & action_marker : StaticActionMarkers )
+        {
+            auto marker = FSFSplineMarker( action_marker.Name, action_marker.ItIsEnabled, action_marker.Infos, NewObject< USFSplineMarkerObject_Action >( this ) );
+            Cast< USFSplineMarkerObject_Action >( marker.Object )->ActionClass = action_marker.ActionClass->GetOwnerClass();
+            SplineMarkers.Add( marker );
+        }
+
+        StaticActionMarkers.Empty();
+
+        for ( const auto & level_actor_marker : LevelActorActionMarkers )
+        {
+            auto marker = FSFSplineMarker( level_actor_marker.Name, level_actor_marker.ItIsEnabled, level_actor_marker.Infos, NewObject< USFSplineMarkerObject_LevelActor >( this ) );
+            Cast< USFSplineMarkerObject_LevelActor >( marker.Object )->LevelActor = level_actor_marker.LevelActor;
+            SplineMarkers.Add( marker );
+        }
+
+        LevelActorActionMarkers.Empty();
+
+        for ( const auto & data_marker : DataMarkers )
+        {
+            auto marker = FSFSplineMarker( data_marker.Name, data_marker.ItIsEnabled, data_marker.Infos, NewObject< USFSplineMarkerObject_Data >( this ) );
+            Cast< USFSplineMarkerObject_Data >( marker.Object )->Data = data_marker.Data;
```

Comment:
3 times almost the same code. you could create one implementation that would look like (not tested):

```
const auto create_marker = []< typename T, typename U >( T legacy_marker ){
    auto * marker_object = NewObject< U >( this );
    return new_marker( legacy_marker.Name, legacy_marker.ItIsEnabled, legacy_marker.Infos, marker_object );
    
};

for ( const auto & data_marker : DataMarkers )
{
    auto new_marker = create_marker< FSFSplineMarker_Data, USFSplineMarkerObject_Data>( data_marker );
    new_marker->Data = data_marker.Data;
    SplineMarkers.Emplace( new_marker );
}
```

Reply:
Couldn't create the object in the lambda